app/messaging/agreement-changed.js

```diff
@@ -24,25 +24,33 @@ function selectExpression (ctx) {
       return calcs[i].expression
     }
   }
-  log('No conditions were matched to provide an expression. Check the coverage of the conditions. Default expression will be used.')
-  return 'userInput * paymentRate'
+  throw new Error('No conditions were matched to provide an expression. Check the coverage of the conditions. Default expression will be used.')
 }
 
-function calcPayment (standard) {
-  const exp = selectExpression(standard)
-  log(`Calculation expression used for '${standard.id}' - '${exp}'`)
-  return evalExpression(standard, exp)
+/**
+ * @param {Object} ctx - An object with an `id` prop and a `calculations` prop
```

Comment:
I've added this here for a bit of a test to see if it helps with understanding the usage of it. If we working with a typed language I think the need would be reduced, potentially completely.

Reply:
I thought writing useful comments like function signatures was frowned upon by the javascript community, just wasted keystrokes 😉 

`<serious-comment>`I have no issue with useful comments`</serious-comment>`